test/database/dao/BookingDataDAOTest.java

```diff
@@ -16,9 +18,79 @@ public class BookingDataDAOTest {
+	public void testBookingCrudTest() throws SQLException {
+		LocalDateTime entryDate=LocalDateTime.now().withNano(0);
 		
-		 bookingDataDAO.save(booking);		
+		BookingDataDTO booking=new BookingDataDTO(entryDate,LocalDateTime.of(2023,12,12,10,30),PaymentMethodDTO.CASH,new BigDecimal("6000.0"));
+		
+		BookingDataDTO bookingSaved = bookingDataDAO.save(booking);
+		Assert.assertNotNull(bookingSaved);
+		Assert.assertNotNull(bookingSaved.getId());
+		
+		BookingDataDTO bookingSearched = bookingDataDAO.searchByIdBooking(bookingSaved.getId());
+		Assert.assertNotNull(bookingSearched);
+		assertAttributes(bookingSaved,bookingSearched);
+		
+		BookingDataDTO bookingToModify=new BookingDataDTO(bookingSearched.getId(),LocalDateTime.of(2023,12,10,10,30), LocalDateTime.of(2024,01,01,10,30), new BigDecimal("30000.0"), PaymentMethodDTO.CREDIT);
+		
+		int bookingModified = bookingDataDAO.modify(bookingToModify);
+		Assert.assertNotNull(bookingSearched);
+		Assert.assertTrue(1 == bookingModified);
+		
+		BookingDataDTO bookingSearchedModified = bookingDataDAO.searchByIdBooking(bookingToModify.getId());
+		Assert.assertNotNull(bookingSearchedModified);
+		assertAttributes(bookingToModify,bookingSearchedModified);
+		
+		int deletedRecords = bookingDataDAO.delete(bookingToModify.getId());
+		Assert.assertTrue(1 == deletedRecords);
+		
+		BookingDataDTO bookingDeleted = bookingDataDAO.searchByIdBooking(bookingToModify.getId());
+		Assert.assertNull(bookingDeleted);
+		
+	}
+
+	@Test
+	public void testSearchBookingList() {
+		LocalDateTime entryDate=LocalDateTime.now().withNano(0);
+		BookingDataDTO booking=new BookingDataDTO(entryDate,LocalDateTime.of(2023,12,12,11,30),PaymentMethodDTO.CASH,new BigDecimal("6000.0"));
+		BookingDataDTO booking2=new BookingDataDTO(entryDate,LocalDateTime.of(2023,10,12,8,30),PaymentMethodDTO.CASH,new BigDecimal("7000.0"));
+		BookingDataDTO booking3=new BookingDataDTO(entryDate,LocalDateTime.of(2023,8,12,07,30),PaymentMethodDTO.CASH,new BigDecimal("8000.0"));
+		
+		BookingDataDTO bookingSaved = bookingDataDAO.save(booking);
+		BookingDataDTO bookingSaved1 = bookingDataDAO.save(booking2);
+		BookingDataDTO bookingSaved2 = bookingDataDAO.save(booking3);
+		
+		List<BookingDataDTO> searchBookingList = bookingDataDAO.searchBookingList();
+		assertAttributes(bookingSaved,searchBookingList.get(0));
+		assertAttributes(bookingSaved1,searchBookingList.get(1));
+		assertAttributes(bookingSaved2,searchBookingList.get(2));
+		
+		int deletedRecords = bookingDataDAO.delete(bookingSaved.getId());	
+		int deletedRecords2 = bookingDataDAO.delete(bookingSaved1.getId());
+		int deletedRecords3 = bookingDataDAO.delete(bookingSaved2.getId());
+		Assert.assertTrue(1 == deletedRecords3);
+		Assert.assertTrue(1 == deletedRecords2);
+		Assert.assertTrue(1 == deletedRecords);
+	}	
+	
+	@Test
+	public void testGetBookingAttributes() {
+		Assert.assertThrows(RuntimeException.class,()->{
```

Comment:
Let's do this test in a different way, the idea is to be able to assert over the exception message, something that I can't see the `Assert.throws` currently allows.

```java
			try {
				LocalDateTime entryDate = LocalDateTime.now().withNano(0);
				BookingDataDTO booking = new BookingDataDTO(null, null, null, new BigDecimal("6000.0"));
				BookingDataDTO booking2 = new BookingDataDTO(entryDate, LocalDateTime.of(2023, 10, 12, 8, 30), PaymentMethodDTO.CASH, new BigDecimal("7000.0"));

				bookingDataDAO.save(booking);
				bookingDataDAO.save(booking2);
				Assert.fail("This test should have failed");
			} catch (RuntimeException ex) {
				// Assert ex is not null
				// assert ex.getMessage() has the expected message
			}
```